src/util/async/context/SyncExecutionContext.hpp

```diff
@@ -56,6 +56,24 @@ struct SameThreadContext {
 
     Executor executor;
 
+    void
+    stop() const noexcept
+    {
+        executor.stop();
+    }
+
+    void
+    join() noexcept
+    {
+        executor.join();
```

Comment:
I think Executor's join can be updated to const. Then we can change join interface to const too.

Reply:
In the very deep end, `boost::asio::thread_pool::join` is not const, so I don't think we can.

Reply:
I tried and turns out it still compiles fine. Strange but let's roll with it 👍